### DISCUSSION TITLE
DBNet customizing

### DISCUSSION THREAD

**Discussion by simanast** [open-mmlab/mmocr#1536] [Q&A]
I'm trying to customize pretrained model `DB_r50`. From what i know, DBNet should be able to process images of any shape. But in model's config [file](https://github.com/open-mmlab/mmocr/blob/main/configs/_base_/det_pipelines/dbnet_pipeline.py) i found `EastRandomCrop` to (640, 640). Whenever i try to delete it and replace with `Resize`, the pipeline won't work, i get a warning like below for every file in my dataset :
```
prepare index 343 with error 
Warning: skip broken file {...}
```
But `EastRandomCrop` to different shape works fine.
So i wonder how can i make DBNet train on resized images, not cropped.

**Comment by simanast** [comment]
Or is `EastRandomCrop` an augmentation in this case? Since it's not wrapped in such dict (like `ImgAug`) i'm not sure

**Comment by gaotongxiao** [comment]
In MMOCR 0.x, the augmentations do not necessarily follow the same protocol and thus may not be interchangeable. Hence, simply replacing one transform with another could result in errors.

In MMOCR 1.0, `EastRandomCrop` has been replaced by three atomic transforms, and you can safely modify the order or change the parameters of these transforms by your need. (See [DBNet's config](https://github.com/open-mmlab/mmocr/blob/3433c8cba474e1f53340183a6648c64813c3aa78/configs/textdet/dbnet/_base_dbnet_resnet18_fpnc.py#L49-L51) and [Migration Guide](https://mmocr.readthedocs.io/en/dev-1.x/migration/transforms.html#data-augmentation-related-data-transforms) for details]. I'd suggest you upgrade to MMOCR 1.0 to carry out more solid investigations.
